### PR TITLE
Feature/accounts resolver

### LINKED ISSUE #7: Create user Resolver
#Expected Behaviour
A user should be able to sign in/ sign up

##Steps to Reproduce the problem
This is basically enabling communication between the client end and the server, we can achieve this by:
* [ ] creating user resolvers
* [ ] creating user mutations

### PR DESCRIPTION
**Pull request by WanjikuMac**
Fixes #7

### CODE REVIEW

File: lib/hotel_web/schema/middleware/changeset_errors.ex [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+        |> Enum.map(fn {item, error} ->"#{item}: #{error}" end)
+  end
+    #handle other kind of errors
+    defp transform_error(error), do: [error]
+
+    #build error using options
+    @spec build_error(tuple()) :: String.t()
+    defp build_error({msg, opts}) do
+      Enum.reduce(opts, msg, fn({key, value}, acc) ->
+        String.replace(acc, "%{#{key}}", to_string(value))
```
Comment (zacck-zz) [inline lib/hotel_web/schema/middleware/changeset_errors.ex:30]: @WanjikuMac  this error could include more details usually in an `opts` field

Comment (WanjikuMac) [reply lib/hotel_web/schema/middleware/changeset_errors.ex:30]: @zacck Could you please elaborate or do you mean something like this? And is it implemented in the `@behavoir` functions of `Absinthe.Middleware`
```defp build_error({msg, opts(attrs)}) do```

Comment (zacck-zz) [reply lib/hotel_web/schema/middleware/changeset_errors.ex:30]: I mean the `opts` item has more details usually about the error that the user can use to recover

Comment (WanjikuMac) [reply lib/hotel_web/schema/middleware/changeset_errors.ex:30]: @zacck Oh okay, how would I then can I added a detailed error msg with even a couple of solutions?

Comment (zacck-zz) [reply lib/hotel_web/schema/middleware/changeset_errors.ex:30]: like this https://github.com/LittleKidogo/quick_bytes_backend/blob/3f8c1d75b26c5835f70eed3a464351e806a104d9/lib/qb_backend_web/schema/middleware/changeset_errors.ex#L30

Comment (WanjikuMac) [reply lib/hotel_web/schema/middleware/changeset_errors.ex:30]: @zacck Which is the same thing I have ama?

**Comment by WanjikuMac** [comment]
@zacck final tests for this PR, your review will be highly appreciated

**Comment by zacck-zz** [comment]
@WanjikuMac  looks awesome, coming well keep going :)

**Pull request merged**